### PR TITLE
Improvements, adding RTM option in bot options

### PR DESCRIPTION
**Pull request by seriouslysean**
Various cleanup and some RTM (Real Time Messaging) requirements. Comments inline!

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -53,9 +53,9 @@ This bot demonstrates many of the core features of Botkit:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
 
 // Only request env file if it exists
-var fs = require('fs');
-var env = require('node-env-file');
-var env_file = __dirname + '/.env';
+const fs = require('fs');
+const env = require('node-env-file');
+const env_file = __dirname + '/.env';
```
Comment (seriouslysean) [inline bot.js:58]: Converted a lot of the vars to consts for security.

File: bot.js [hunk 2]
```diff
@@ -65,110 +65,68 @@ if (!process.env.clientId || !process.env.clientSecret || !process.env.PORT) {
...
+const bot_options = {
     clientId: process.env.clientId,
     clientSecret: process.env.clientSecret,
     clientSigningSecret: process.env.clientSigningSecret,
     // debug: true,
     scopes: ['bot'],
     studio_token: process.env.studio_token,
-    studio_command_uri: process.env.studio_command_uri
+    studio_command_uri: process.env.studio_command_uri,
+    require_delivery: true, // required for RTM (Real Time Messaging) support
```
Comment (seriouslysean) [inline bot.js]: This is required if we ever use RTM, it makes sure a message is delivered before the bot actions on it and ensures things happen in order.

File: bot.js [hunk 3]
```diff
@@ -65,110 +65,68 @@ if (!process.env.clientId || !process.env.clientSecret || !process.env.PORT) {
...
 controller.startTicking();
 
 // Set up an Express-powered webserver to expose oauth and webhook endpoints
-var webserver = require(__dirname + '/components/express_webserver.js')(controller);
+const webserver = require(__dirname + '/components/express_webserver.js')(controller);
 
-if (!process.env.clientId || !process.env.clientSecret) {
+if (process.env.clientId && process.env.clientSecret) {
 
-  // Load in some helpers that make running Botkit on Glitch.com better
```
Comment (seriouslysean) [inline bot.js:100]: Removed all the Glitch.com stuff we'll never use.

File: bot.js [hunk 4]
```diff
@@ -65,110 +65,68 @@ if (!process.env.clientId || !process.env.clientSecret || !process.env.PORT) {
...
-  console.log('WARNING: This application is not fully configured to work with Slack. Please see instructions at ' + where_its_at);
-}else {
-
-  webserver.get('/', function(req, res){
-    res.render('index', {
-      domain: req.get('host'),
-      protocol: req.protocol,
-      glitch_domain:  process.env.PROJECT_DOMAIN,
-      layout: 'layouts/default'
+    webserver.get('/', (req, res) => {
```
Comment (seriouslysean) [inline bot.js:101]: If we have a `clientId` and `clientSecret`, just show the install page.

File: bot.js [hunk 5]
```diff
@@ -65,110 +65,68 @@ if (!process.env.clientId || !process.env.clientSecret || !process.env.PORT) {
...
+
+    const normalizedPath = require("path").join(__dirname, "skills");
 
+    require("fs").readdirSync(normalizedPath).forEach((file) => {
+        require("./skills/" + file)(controller);
+    });
 
+} else {
 
+    webserver.get('/', (req, res) => {
```
Comment (seriouslysean) [inline bot.js:125]: If we don't have a `clientId` or `clientSecret`, just show a very basic message.

File: bot.js [hunk 6]
```diff
@@ -65,110 +65,68 @@ if (!process.env.clientId || !process.env.clientSecret || !process.env.PORT) {
...
+const debug = require('debug')('botkit:main');
 
-var bot_options = {
+const bot_options = {
     clientId: process.env.clientId,
     clientSecret: process.env.clientSecret,
     clientSigningSecret: process.env.clientSigningSecret,
     // debug: true,
     scopes: ['bot'],
     studio_token: process.env.studio_token,
```
Comment (dgautsch) [inline bot.js]: remove studio tokens since we aren't going to use that

**Review by dgautsch — APPROVED** [review]

**Pull request merged**